contracts/core/Allo.sol

```diff
@@ -33,57 +57,108 @@ contract Allo is Initializable {
     // takes fee from user
     // validates that the owner is actually allowed to use the identity
     function createPool(
-        PoolData memory /*_poolData*/
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address _distributionStrategy,
+        Metadata _metadata
     ) external pure returns (uint) {
-        uint32 _poolId = 0;
 
-        // todo: return the poolId? what do we want to return here?
-        return _poolId;
+        if (!registry.isMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        
+        // ASK should we clone _allocationStrategy / _distributionStrategy or just use them directly?
+        // bytes32 salt = keccak256(abi.encodePacked(msg.sender, nonce, "allocationStrategy"));
+        // address clone = ClonesUpgradeable.cloneDeterministic(_allocationStrategy, salt);
+        
+        Pool memory pool = Pool(
+            _identityId,
+            _allocationStrategy,
+            _distributionStrategy,
+            _metadata,
+            false
+        );
+
+        // ASK: Does this function also recieve funds ?
```

Comment:
@zobront

Reply:
We could add the two params we need (`_poolId` & `_amount`) as optional to let the user add funds on pool creation and call the `fundPool()` function.

```javascript
function fundPool(uint _poolId, uint _poolAmt) external payable {
    // ASK: what do we do with _poolAmt ?
}

// Call fundPool() from createPool() if they pass any tokens
function createPool(
    bytes32 _identityId,
    address _allocationStrategy,
    address _distributionStrategy,
    Metadata _metadata,
    uint _poolId,
    uint _amount
) external pure returns (uint poolId) {

    if (!registry.isMemberOfIdentity(_identityId, msg.sender)) {
        revert NO_ACCESS_TO_ROLE();
    }

    Pool memory pool = Pool(
        _identityId,
        IAllocationStrategy(_createClone(_allocationStrategy)),
        IDistributionStrategy(_createClone(_distributionStrategy)),
        _metadata,
        false
    );

    // check for tokens passed in and call fundPool() if _amount > 0
    fundPool(_poolId, _amount);

    // ASK: Does this function also recieve funds ?
    // If so -> ETH / ERC20 ?
    // Should we also accept amount as an argument to check ?

    poolId = _poolIndex++;
    pools[poolId] = pool;

    emit PoolCreated(
        poolId,
        _identityId,
        _allocationStrategy,
        _distributionStrategy,
        _metadata
    );
}
```

Reply:
Yup good callout. Added those